integration/spark/shared/src/test/java/io/openlineage/spark/agent/util/PathUtilsTest.java

```diff
@@ -180,6 +180,26 @@ void testFromCatalogWithDefaultStorage() throws URISyntaxException {
     assertThat(di.getSymlinks().get(0).getNamespace()).isEqualTo("/warehouse");
   }
 
+  @Test
+  void testFromCatalogWithDefaultStorageAndNoWarehouse() throws URISyntaxException {
+    sparkConf.remove("spark.hadoop.hive.metastore.uris");
+    when(catalogTable.storage()).thenReturn(catalogStorageFormat);
+    when(catalogStorageFormat.locationUri())
+        .thenReturn(Option.apply(new URI("s3://s3-db/table")));
+
+    TableIdentifier tableIdentifier = mock(TableIdentifier.class);
+    when(catalogTable.identifier()).thenReturn(tableIdentifier);
+    when(tableIdentifier.database()).thenReturn(Option.apply("db"));
+    when(tableIdentifier.table()).thenReturn("table");
+
+    DatasetIdentifier di = PathUtils.fromCatalogTable(catalogTable, Optional.of(sparkConf));
+    assertThat(di.getName()).isEqualTo("table");
+    assertThat(di.getNamespace()).isEqualTo("s3://s3-db");
+    assertThat(di.getSymlinks()).hasSize(1);
+    assertThat(di.getSymlinks().get(0).getName()).isEqualTo("db.table");
+    assertThat(di.getSymlinks().get(0).getNamespace()).isEqualTo("");
```

Comment:
A reference backend implementation creates lineage graph based on symlinks as well. Thus it requires uniqueness of the symlinks. In this case, which is a spark table without megastore URI, we want to use send table name within the symlink, although we must still enforce its uniqueness and cannot leave empty `namespace`. That's why `s3://s3-db` as it provides the wanted guarantees.

Reply:
Yes I agree with you.
But with the current code we get
symlinks[0].namespace=feat_count
where feat_count is the name of the table.

Reply:
Yes, I think `name` should be a table name. `namespace` should point to metastore URI if it is present. Otherwise this can be warehouse location within the bucket. In the example you provide warehouse location is just a bucket root.

Reply:
Do you mean by bucket root just a "/" ?

Reply:
For now we have a namespace table in the database full of table names and it is due to `StringUtils.substringBeforeLast(di.getName(), File.separator)`.

In our case each team has its own bucket and its own database.

Some chose to write at the root of the bucket and some chose to write in some path. Both use the Glue Catalog.

For us there is no notion of warehouse. 

What do you think could be a good solution ?

Reply:
If I follow what you said, in these examples :

1. db_team1.table1 -> (s3://db_team1_bucket/table1)
2. db_team2.table2 -> (s3://db_team2_bucket/folder2/table2)

Currently 
- in the 1st example, the namespace of the symlink is `table1`
- in the 2nd example, the namespace of the symlink is `/folder2`

Do you think we should have ?
- `symlinks[0].namespace="/"` instead of `table1`
- `symlinks[0].namespace="/folder2"`

If it is the case, I can implement it.

(Knowing that in our case, the number of namespace such as `/folder2` can be very high. Almost as much as the number of tables).

Reply:
For `when(catalogStorageFormat.locationUri()).thenReturn(Option.apply(new URI("s3://s3-db/table")));`
it should be:
```
assertThat(di.getName()).isEqualTo("table");
assertThat(di.getNamespace()).isEqualTo("s3://s3-db");
assertThat(di.getSymlinks().get(0).getName()).isEqualTo("db.table");
assertThat(di.getSymlinks().get(0).getNamespace()).isEqualTo("s3://s3-db");
```

`when(catalogStorageFormat.locationUri()).thenReturn(Option.apply(new URI("s3://s3-db/some-path/table")));`
it should be:
```
assertThat(di.getName()).isEqualTo("/some-path/table");
assertThat(di.getNamespace()).isEqualTo("s3://s3-db");
assertThat(di.getSymlinks().get(0).getName()).isEqualTo("db.table");
assertThat(di.getSymlinks().get(0).getNamespace()).isEqualTo("s3://s3-db/some-path/");
```

does it make sense?